### ISSUE TITLE
Issues with being able create multiple user defined functions with same name and signature

### ISSUE THREAD

**Issue by dfgitn4j** [neo4j-contrib/neo4j-apoc-procedures#1558]
## Expected Behavior (Mandatory)

Allow only one function name / signature combination to be created

## Actual Behavior (Mandatory)
Creates duplicates, can't remove them

## How to Reproduce the Problem

`apoc.custom.asFunction` allows duplicate function name / signature combinations.  Error when try and remove with `apoc.custom.removeFunction`

### Simple Dataset (where it's possibile)

```
CALL apoc.custom.asFunction('tooManyLifeUniverseEverythings','RETURN 42');
CALL apoc.custom.asFunction('tooManyLifeUniverseEverythings','RETURN 42');

```

### Steps (Mandatory)

  1. create same function twice with apoc.custom.asFunction
  1. try and remove function(s) with apoc.custom.removeFunction
 

## Screenshots (where it's possibile)

```
CALL apoc.custom.list
╒══════════╤════════════════════════════════╤═════════════╤══════╤═══════════╤═══════════════════════╤══════════════╤═════════════╕
│"type"    │"name"                          │"description"│"mode"│"statement"│"inputs"               │"outputs"     │"forceSingle"│
╞══════════╪════════════════════════════════╪═════════════╪══════╪═══════════╪═══════════════════════╪══════════════╪═════════════╡
│"function"│"tooManyLifeUniverseEverythings"│""           │null  │"RETURN 42"│[["params","map","{}"]]│"list? of map"│false        │
├──────────┼────────────────────────────────┼─────────────┼──────┼───────────┼───────────────────────┼──────────────┼─────────────┤
│"function"│"tooManyLifeUniverseEverythings"│""           │null  │"RETURN 42"│[["params","map","{}"]]│"list? of map"│false        │
└──────────┴────────────────────────────────┴─────────────┴──────┴───────────┴───────────────────────┴──────────────┴─────────────┘
```

Gives
```
call apoc.custom.removeFunction('tooManyLifeUniverseEverythings')

Neo.ClientError.Procedure.ProcedureCallFailed
Failed to invoke procedure `apoc.custom.removeFunction`: Caused by: java.util.NoSuchElementException: More than one element in java.util.Spliterators$1Adapter@235d2118. First element is 'Node[28]' and the second element is 'Node[30]'

```

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: Catalina 10.5.15
  - Neo4j: 3.x 4.x
  - Neo4j-Apoc: apoc.custom.asFunction

**Comment by augonis** [comment]
This has same cause as (and fix) #1555

**Comment by conker84** [comment]
closed as duplicate of #1555
